### DISCUSSION TITLE
It was hard to find what I need in the docs

### DISCUSSION THREAD

**Discussion by devton-here** [Homebrew/discussions#3314] [Everyday usage]
After finishing my first formula for a binary CLI (as a bottle) in my own repository (tap), I would like to give a short feedback.

For me it was hard to find some important concepts in the documentation. On the other side the community response (in this Discussions space) was super fast and helpful.

some more details:

- missing menu (except on the main page) on https://docs.brew.sh/ but it's not common for me to navigate back and forth from it
- it took some time to understand that it doesn't matter what's behind the `url`, both is okay: a source or binary archive
- it was not clear to me that Homebrew is automatically (downloading and) and extracting the archive from the `url`
- I did not found anything about how to setup a jar (`write_jar_script`)

**Comment by SMillerDev** [comment]
> it took some time to understand that it doesn't matter what's behind the url, both is okay: a source or binary archive

Homebrew will download whatever you define as the URL but Formula files define a build so if it's downloading a binary archive it's the wrong type of Homebrew package (a cask would be the good for those).
The Formula cookbook also defines this: https://docs.brew.sh/Formula-Cookbook#grab-the-url

> it was not clear to me that Homebrew is automatically (downloading and) and extracting the archive from the url

Pull requests are welcome although that seems like a pretty standard feature for a package manager.

> I did not found anything about how to setup a jar (write_jar_script)

Anything that's specific to a certain build system likely changes to quick to keep up with in documentation. The best option is to just look for prior art.

---

https://github.com/Homebrew/brew/tree/master/docs and https://github.com/Homebrew/formulae.brew.sh/ are eagerly awaiting your pull requests.
